### DISCUSSION TITLE
Evaluating Unity3D GLTF plugins for adding OMI extensions

### DISCUSSION THREAD

**Discussion by mikeskydev** [omigroup/gltf-extensions#11] [General]
Making a new discussion thread to document and evaluate the state of the various Unity-based GLTF plugins. They are all at various states in terms of what features they support. This may depend on the end goal of the plugin for each of the plugin's authors. I think this should be taken into consideration during the evaluation, but first I think we should objectively examine the features they have currently.

We need to define what criteria we will use for an objective evaluation. We should keep in mind the target audience for our plugins during this process. #3 

To start off, but welcome to suggestions:

## Extension support
- [ ] Native extension support

As our target is new GLTF extensions, one of the main questions to ask. Does the plugin have an out-of-the-box method to add any arbitrary processor of GLTF extensions? We may also want to consider the feasibility of adding this functionality to the plugin if it does not exist. Another option is to directly add OMI extensions to a plugin on a fork or pull request, though this may limit extendibility, and again may depend on the plugin author's end goals.

### Editor
- [ ] Import
- [ ] Export
### Runtime
- [ ] Import
- [ ] Export

How can the plugin be used to process the gltf format? Some users may use Unity as a step in their asset authoring pipeline rather than it being the final presentation medium.

## Platform support
- [ ] Windows
- [ ] macOS
- [ ] Linux
- [ ] iOS
- [ ] Android
- [ ] WebGL

Unity supports a number of different platforms which can sometimes require different plugins. Some plugins may work only on a subset of Unity's supported platforms.

## Render pipelines
- [ ] Standard Render Pipeline
- [ ] Universal Render Pipeline
- [ ] High Definition Render Pipeline

Unity currently has 3 render pipelines, sometimes with different shaders required for each.

**Comment by mikeskydev** [comment]
# glTFast

[Repository](https://github.com/atteneder/glTFast)

More feature information [here](https://github.com/atteneder/glTFast/blob/main/Documentation~/features.md)

## Extension support
- [ ] Native extension support (Issue atteneder/glTFast#226)

### Inbuilt supported extensions:
`KHR_draco_mesh_compression`
`KHR_materials_pbrSpecularGlossiness`
`KHR_materials_unlit`
`KHR_texture_transform`
`KHR_mesh_quantization`
`KHR_texture_basisu`
`EXT_mesh_gpu_instancing`

## Asset pipeline stages
### Editor
- [x] Import
- [ ] Export (Draft PR atteneder/glTFast#249, Implemented)
### Runtime
- [x] Import
- [ ] Export (Draft PR atteneder/glTFast#249, Proposed)

## Platform support
- [x] Windows
- [x] macOS
- [x] Linux
- [x] iOS
- [x] Android
- [x] WebGL
- [x] UWP
- [x] Lumin

## Render pipelines
- [x] Standard Render Pipeline
- [x] Universal Render Pipeline
- [x] High Definition Render Pipeline

**Reply by mikeskydev** [reply]
Updated with inbuilt extensions

**Comment by mikeskydev** [comment]
# UniGLTF (UniVRM)

[Repository](https://github.com/vrm-c/UniVRM)

More feature information [here](https://github.com/vrm-c/UniVRM/tree/master/Assets/UniGLTF)

## Extension support
- [x] Native extension support

### Inbuilt supported extensions

`KHR_materials_unlit`
`KHR_texture_transform`

## Asset pipeline stages
### Editor
- [x] Import
- [x] Export
### Runtime
- [x] Import
- [ ] Export

## Platform support
- [x] Windows
- [x] macOS
- [x] Linux
- [ ] iOS
- [ ] Android
- [ ] WebGL

## Render pipelines
- [x] Standard Render Pipeline
- [ ] Universal Render Pipeline (Experimental support)
- [ ] High Definition Render Pipeline (No support)

**Reply by 0b5vr** [reply]
It seems this page explains how to implement extensions for UniGLTF (I didn't know it supports!)

https://github.com/vrm-c/UniVRM/blob/4ae9b94e89ede7d1eb481801e356f3491b813629/docs/gltf/how_to_impl_extension.md

**Reply by mikeskydev** [reply]
Thanks, updated!

**Reply by mikeskydev** [reply]
Updated with inbuilt extensions
